src/content/docs/en/guides/cms/builderio.mdx

```diff
@@ -3,19 +3,503 @@ title: Builder.io & Astro
+3. In the panel on the right under the **options** tab, you should see the data fields `title` and `slug`. These are the fields you've created when you [set up your model](#creating-a-model-for-a-blog-post). Fill them in with the following values:
+    - **title:** 'First post, woohoo!'
+    - **slug:** 'first-post-woohoo'
+4. Design your post. Click on the **Add Block** button and add something - maybe a text field with some text to start with.
+5. In the text field above the editor, give your entry a name. This is how it will be listed in the Builder app.
+6. When you're ready click the **Publish** button in the upper right corner.
+
+Your content is now live in the Builder CMS but not part of your Astro project yet. You'll be fetching and displaying it in the following steps.
+
+### Displaying a list of blog posts
+
+Create the page `src/pages/index.astro` that will display a list of post titles, each linking to its own page:
+
+<FileTree title="Project Structure">
+- src/
+  - pages/
+    - **index.astro**
+  - env.d.ts
+- .env
+- astro.config.mjs
+- package.json
+</FileTree>
+
+Then add the following content:
+
+```astro title="src/pages/index.astro" /title|slug/
+---
+const builderAPIpublicKey = import.meta.env.BUILDER_API_PUBLIC_KEY;
+const builderModel = import.meta.env.BUILDER_BLOGPOST_MODEL;
+
+const { results: posts } = await fetch(
+  `https://cdn.builder.io/api/v3/content/${builderModel}?${new URLSearchParams({
+    apiKey: builderAPIpublicKey,
+    fields: ["data.slug", "data.title"].join(","),
+    cachebust: "true",
+  }).toString()}`
+)
+  .then((res) => res.json())
+  .catch();
+---
+```
+
+Fetching via the content API returns an array of objects containing data for each post. The `fields` query parameter tells Builder which data is included (see highlighted code). `slug` and `title` should match the names of the custom data fields you've added to [your Builder model](#creating-a-model-for-a-blog-post).
+
+
+Use the `posts` array returned from the fetch to display a list of blog posts on the page:
+
+```astro title="src/pages/index.astro" ins={18-99}
+---
+const builderAPIpublicKey = import.meta.env.BUILDER_API_PUBLIC_KEY;
+const builderModel = import.meta.env.BUILDER_BLOGPOST_MODEL;
+
+const { results: posts } = await fetch(
+  `https://cdn.builder.io/api/v3/content/${builderModel}?${new URLSearchParams({
+    apiKey: builderAPIpublicKey,
+    fields: ["data.slug", "data.title"].join(","),
+    cachebust: "true",
+  }).toString()}`
+)
+  .then((res) => res.json())
+  .catch();
+---
+
+<html lang="en">
+  <head>
+    <title>Blog Index</title>
+  </head>
+  <body>
+    <ul>
+      {
+        posts.map(({ data: { slug, title } }) => (
+          <li>
+            <a href={`/posts/${slug}`}>{title}</a>
+          </li>
+        ))
+      }
+    </ul>
+  </body>
+</html>
+
+```
```

Comment:
``````suggestion
```astro title="src/pages/index.astro" ins={18-99}
---
const builderAPIpublicKey = import.meta.env.BUILDER_API_PUBLIC_KEY;
const builderModel = import.meta.env.BUILDER_BLOGPOST_MODEL;

const { results: posts } = await fetch(
  `https://cdn.builder.io/api/v3/content/${builderModel}?${new URLSearchParams({
    apiKey: builderAPIpublicKey,
    fields: ["data.slug", "data.title"].join(","),
    cachebust: "true",
  }).toString()}`
)
  .then((res) => res.json())
  .catch();
---

<html lang="en">
  <head>
    <title>Blog Index</title>
  </head>
  <body>
    <ul>
      {
        posts.map(({ data: { slug, title } }) => (
          <li>
            <a href={`/posts/${slug}`}>{title}</a>
          </li>
        ))
      }
    </ul>
  </body>
</html>

```

Fetching via the content API returns an array of objects containing data for each post. The `fields` query parameter tells Builder which data is included (see highlighted code). `slug` and `title` should match the names of the custom data fields you've added to your Builder model.


The `posts` array returned from the fetch displays a list of blog post titles on the home page. The individual page routes will be created in the next step.

:::tip[framework integrations]
If you are using a JavaScript framework (e.g. Svelte, Vue, or React) in your Astro project you can use one of Builder's integrations as an alternative to making raw fetch calls through the REST API.
:::

``````
This looks like a big edit, but all I did was swap it so that the code sample comes first, right after the line that says "add the following code." And, here's (somewhere in this section) the space for our tip! Once we figure out exactly where to add it, we should delete it from the top of the page!